crates/loam-cli/src/commands/build/clients.rs

```diff
@@ -254,94 +259,115 @@ export default new Client.Client({{
-                let hash = cli::contract::install::Cmd::parse_arg_vec(&[
-                    "--wasm",
-                    wasm_path
-                        .to_str()
-                        .expect("we do not support non-utf8 paths"),
-                ])?
-                .run_against_rpc_server(None, None)
-                .await?
-                .into_result()
-                .expect("no hash returned by 'contract install'")
-                .to_string();
-                eprintln!("    ↳ hash: {hash}");
-
-                // Check if we have an alias saved for this contract
-                let alias = Self::get_contract_alias(name)?;
-                if let Some(contract_id) = alias {
-                    match self
-                        .contract_hash_matches(&contract_id, &hash, network)
-                        .await
-                    {
-                        Ok(true) => {
-                            eprintln!("✅ Contract {name:?} is up to date");
-                            continue;
-                        }
-                        Ok(false) if self.loam_env(LoamEnv::Production) == "production" => {
-                            return Err(Error::ContractUpdateNotAllowed(name.to_string()));
-                        }
-                        Ok(false) => eprintln!("🔄 Updating contract {name:?}"),
-                        Err(e) => return Err(e),
-                    }
+            }
+        }
+        for name in package_names {
+            let settings = match contracts {
+                Some(contracts) => contracts.get(&name as &str),
+                None => None,
+            };
+            // Skip only if contract is found and its workspace setting is false
+            if let Some(c) = settings {
+                if !c.workspace {
+                    eprintln!("Skipping {name:?} as its workspace setting is false");
+                    continue;
                 }
-
-                eprintln!("🪞 instantiating {name:?} smart contract");
-                let contract_id = cli::contract::deploy::wasm::Cmd::parse_arg_vec(&[
-                    "--alias",
-                    name,
-                    "--wasm-hash",
-                    &hash,
-                ])?
-                .run_against_rpc_server(None, None)
-                .await?
-                .into_result()
-                .expect("no contract id returned by 'contract deploy'");
-                eprintln!("    ↳ contract_id: {contract_id}");
-
-                // Save the alias for future use
-                Self::save_contract_alias(name, &contract_id, network)?;
-
-                // Run init script if we're in development or test environment
-                if self.loam_env(LoamEnv::Production) == "development"
-                    || self.loam_env(LoamEnv::Production) == "testing"
+            } else {
+                eprintln!("Contract {name:?} not found in contracts, but continuing anyway");
```

Comment:
Having trouble following this new logic. This happens if... you skip explicitly listing the contract in `environments.toml`, but it's part of the Cargo workspace?

If that's true, I don't think we need to say anything here. Saying "not found, but..." makes it sound like an error-state, but this will be the default. The primary happy path. Most people will only see output for this scenario, most of the time.

Reply:
Agreed, that was something I meant to take out before the PR